fee-monitor/src/DynamicChecker.ts

```diff
@@ -0,0 +1,499 @@
+                                const prevDeposit =
+                                    this.nominationDeposits.get(signer.value) || new U64(0);
+                                this.nominationDeposits.set(
+                                    signer.value,
+                                    prevDeposit.plus(additionalDeposit),
+                                );
+                                this.tracerForImmedateSettle.withdraw(signer, additionalDeposit);
+                                break;
+                            }
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private reportError(aggregated: AggregatedErrors) {
+        const positionString = `${aggregated.moment.tag}: ${aggregated.moment.value}`;
+
+        if (aggregated.errors.length > 0) {
+            slack.sendWarning(JSON.stringify(aggregated, null, "    "));
+            const errors = aggregated.errors
+                .map(error => `<li>${JSON.stringify(error)}</li>`)
+                .join("<br />\r\n");
+            email.sendWarning(`
+            <p>${positionString}</p>
+            <ul>${errors}</ul>
+            `);
+        } else {
+            console.log(`${positionString} is Okay`);
+        }
+    }
+
+    private finalizeTerm(blockNumber: number, currentTermValidators: string[]) {
+        this.prevTermData = this.currentTermData;
+        this.currentTermData = {
+            intermediateReward: new CCCTracer(),
+            assembledVotesPerAuthor: new Map(),
+            commitedVotesPerValidator: new Map(),
+        };
+
+        this.parentTermFirstBlockNumber = this.parentTermLastBlockNumber + 1;
+        this.parentTermLastBlockNumber = blockNumber;
+        this.parentTermValidators = currentTermValidators;
+    }
+
+    /**
+     * Distribute total reduced rewards following the rank of collected average votes per round.
+     * @param validators previous term's validator.
+     * @param totalReducedReward reduced reward earned by applying penalty.
+     */
+    private applyAdditionalRewards(validators: string[], totalReducedReward: U64) {
+        const mapValidators: Map<number, string[]> = new Map();
+        for (const validator of validators) {
+            const [missed, proposed] = this.prevTermData.assembledVotesPerAuthor.get(validator) || [
+                0,
+                0,
+            ];
+            if (proposed !== 0) {
+                const ratio = missed / proposed;
+                const list = mapValidators.get(ratio);
+                if (list == null) {
+                    mapValidators.set(ratio, [validator]);
+                } else {
+                    list.push(validator);
+                }
+            }
+        }
+        const sortedValidators = Array.from(mapValidators)
+            .sort(([keyA], [keyB]) => {
+                return keyB - keyA;
```

Comment:
This sorts the array descending order. Since the ratio is `missed / proposed`, the lesser validator should be given more reward. Please check it again.

Reply:
Thanks, I reverted it to `keyA - keyB`